services/user-services.js

```diff
@@ -0,0 +1,56 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+
+const userDao = require('../models/user.dao');
+
+const { validateEmail }  = require('../../util/validators');
+
+
+
+const signUp = async(email, password, firstname, lastname, skintype) => {
+ 
+   await validateEmail(email)
+   
+   const user = await userDao.getuserByEmail(email);
+  
+   if (user){
+    const err = new Error('duplicated email');
+    err.statusCode = 400;
+    throw err;
+  }
+  
+   const hashedPassword = await bcrypt.hash(password, 10);
+   
+   return await userDao.createUser(email, hashedPassword, firstname, lastname, skintype); 
+};
+
+const signIn = async(email, password) =>{
+   const user = await userDao.getuserByEmail(email);
+   
+   const hashedPassword = await bcrypt.hash(password, 10);
+
+   const result = await bcrypt.compare(hashedPassword, user.password.toString());
+  
+   if (!user) {
+    const err = new Error('user does not exist');
+    err.statusCode = 404;
+    throw err;
+   }
+   
+
+   if(!result){
+    const err =new Error('invalid password');
+    err.statusCode = 401;
+    throw err;
+   }
+
+   return jwt.sign({userId: user.id }, progress.env.JWT_SECRET);
```

Comment:
마이너하지만 해당 부분은 지속 언급드린 사항과 동일합니다~!
코드 포맷 자체의 통일성을 위해서도 지속 확인해주세요
```suggestion
   return jwt.sign({userId: user.id}, progress.env.JWT_SECRET);
```